src/virtual_dom/vnode.rs

```diff
@@ -31,6 +31,18 @@ pub enum VNode {
     VRef(Node),
 }
 
+impl VNode {
+    pub fn key(&self) -> Option<String> {
+        match self {
+            VNode::VTag(vtag) => vtag.key.clone(),
+            VNode::VText(_) => None,
+            VNode::VComp(vcomp) => vcomp.key.clone(),
+            VNode::VList(_) => None,
```

Comment:
It would be great if lists supported keys as well!

Reply:
Updated to generate keys using the children.